src/sentry/integrations/jira_server/integration.py

```diff
@@ -53,6 +58,80 @@
 )
 
 
+class InstallationForm(forms.Form):
+    url = forms.CharField(
+        label=_('Jira URL'),
+        help_text=_('The base URL for your Jira Server instance, including the host and protocol.'),
+        widget=forms.TextInput(
+            attrs={'placeholder': 'https://jira.example.com'}
+        ),
+    )
+    verify_ssl = forms.BooleanField(
+        label=_('Verify SSL'),
+        help_text=_('By default, we verify SSL certificates '
+                    'when making requests to your Jira instance.'),
+        widget=forms.CheckboxInput(),
+        required=False,
+        initial=True
+    )
+    client_id = forms.CharField(
+        label=_('Jira Consumer Key'),
+        widget=forms.TextInput(
+            attrs={'placeholder': _(
+                'sentry-consumer-key')}
+        )
+    )
+    private_key = forms.CharField(
+        label=_('Jira Consumer Private Key'),
+        widget=forms.Textarea(
+            attrs={'placeholder': _('--PRIVATE KEY--')}
+        )
+    )
+
+    def clean_url(self):
+        """Strip off trailing / as they cause invalid URLs downstream"""
+        return self.cleaned_data['url'].rstrip('/')
+
+
+class InstallationGuideView(PipelineView):
+    """
+    Display a setup guide for creating an OAuth client in Jira
+    """
+
+    def dispatch(self, request, pipeline):
+        if 'completed_guide' in request.GET:
+            return pipeline.next_step()
+        return render_to_response(
+            template='sentry/integrations/jira-server-config.html',
+            request=request,
+        )
+
+
+class InstallationConfigView(PipelineView):
+    """
+    Collect the OAuth client credentials from the user.
+    """
+
+    def dispatch(self, request, pipeline):
+        if request.method == 'POST':
+            form = InstallationForm(request.POST)
+            if form.is_valid():
```

Comment:
So @markstory what happens here if the `form.is_valid()` returns `False`? Will you get a 500 because it'll hit `render_to_response` without `form` having been initialized?

Reply:
If the form is invalid, the `form` from like 117 will still be in scope.